File: app/src/main/java/net/squanchy/navigation/NavigationModule.java
package net.squanchy.navigation;

import android.app.Activity;

import net.squanchy.injection.ActivityContextModule;

import dagger.Module;
import dagger.Provides;

@Module(includes = {ActivityContextModule.class})
public class NavigationModule {

    @Provides
    DebugActivityIntentFactory debugActivityIntentFactory() {
        return new DebugActivityIntentFactory();
    }

    @Provides
    public Navigator navigator(Activity activity, DebugActivityIntentFactory debugActivityIntentFactory) {
        return new Navigator(activity, debugActivityIntentFactory);
    }
}

File: app/src/main/java/net/squanchy/navigation/NavigationModule.kt
package net.squanchy.navigation

import android.app.Activity
import dagger.Module
import dagger.Provides

@Module
internal class NavigationModule {

    @Provides
    internal fun debugActivityIntentFactory() = DebugActivityIntentFactory()

    @Provides
    fun navigator(activity: Activity, debugActivityIntentFactory: DebugActivityIntentFactory) = Navigator(activity, debugActivityIntentFactory)
}
